utilities/update_version.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+# This script updates the READMEs with the latest non-SNAPSHOT version number and increments the SNAPSHOT 
+# version number in the pom.xml files.
+
+# Precondition: the pom.xml is at a "-SNAPSHOT" version before running this script.  
+# Argument (optional):
+#   $1: new version number for pom.xml files (do not include -SNAPSHOT, that is done automatically).
+#   Providing no argument defaults to incrementing revision number from x.y.z-SNAPSHOT to x.y.z+1-SNAPSHOT
+
+# Example: Suppose that before running this script, the pom reads 7.8.9-SNAPSHOT.  This script will replace 
+# all occurrences of #.#.# with 7.8.9 in the README files and the first occurrence of 7.8.9-SNAPSHOT with 
+# 7.8.10-SNAPSHOT in the pom.xml files.
+
+# Get the previous maven project version.
+CURRENT_SNAPSHOT=$(mvn org.apache.maven.plugins:maven-help-plugin:2.1.1:evaluate -Dexpression=project.version | grep -Ev '(^\[|\w+:)')
+RELEASED_VERSION=${CURRENT_SNAPSHOT%%"-"*}
+DEFAULT_UPDATE="${RELEASED_VERSION%.*}.$((${RELEASED_VERSION##*.}+1))"
+NEW_SNAPSHOT_VERSION=${1:-$DEFAULT_UPDATE}-SNAPSHOT
+
+echo "Changing version to $RELEASED_VERSION in README files"
+echo "Changing version to $NEW_SNAPSHOT_VERSION in pom.xml files"
+
+sed -ri "s/[0-9][0-9]*.[0-9][0-9]*.[0-9][0-9]*/$RELEASED_VERSION/g" README.md
+sed -ri "s/[0-9][0-9]*.[0-9][0-9]*.[0-9][0-9]*/$RELEASED_VERSION/g" gcloud-java/README.md
+sed -ri "s/[0-9][0-9]*.[0-9][0-9]*.[0-9][0-9]*/$RELEASED_VERSION/g" gcloud-java-core/README.md
+sed -ri "s/[0-9][0-9]*.[0-9][0-9]*.[0-9][0-9]*/$RELEASED_VERSION/g" gcloud-java-datastore/README.md
+sed -ri "s/[0-9][0-9]*.[0-9][0-9]*.[0-9][0-9]*/$RELEASED_VERSION/g" gcloud-java-examples/README.md
+sed -ri "s/[0-9][0-9]*.[0-9][0-9]*.[0-9][0-9]*/$RELEASED_VERSION/g" gcloud-java-storage/README.md
+
+sed -i "0,/$CURRENT_SNAPSHOT/s/$CURRENT_SNAPSHOT/$NEW_SNAPSHOT_VERSION/g" pom.xml
```

Comment:
POM changes is the only thing for now that I was expected to be triggered manually (using this script) where the docs above should be updated by same script (invoked by Travis) for site generation.